wgsl/index.bs

```diff
@@ -5085,15 +5085,33 @@ Note: The current function will not resume execution if the called function or
 any descendent called function executes a `discard` statement.
 See [[#discard-statement]].
 
-## Restrictions TODO ## {#function-restriction}
-TODO: *This is a stub*
-
-* Recursion is not permitted. (No cycle in the call graph.)
-* Function call parameters
- * Match type and number
- * Restrictions on pointers
- * Aliasing (?)
+## Restrictions ## {#function-restriction}
 
+* Recursion is not permitted.
+    That is, there must be no cycles in the call graph.
+* A [=vertex=] shader must return the `position` [=built-in output variable|built-in variable=].
+    See [[#builtin-variables]].
+* An entry point must never be the target of a [=function call=].
+* If a function has a return type, it must be a [=plain types|plain type=]
+* A [=formal parameter|function parameter=] of pointer type must be in one of
+    the following storage classes:
+    * [=storage classes/function=]
+    * [=storage classes/private=]
+    * [=storage classes/handle=]
+    * [=storage classes/workgroup=]
+* Each argument of a function call of pointer type must be one of:
+    * An [[#address-of-expr|address-of expression]] of a
+        [[#var-identifier-expr|variable identifier expression]]
+    * A function parameter
+* [SHORTNAME] assumes no aliasing is present in the program. As such,
```

Comment:
Technically I think that we can have aliasing within a function, it is only across function boundaries that we cannot have aliasing, is that right?
e.g.
```
fn foo () {
    X = ...;
    const ptr: ptr<...> = &X;
    *ptr = ...;
}
```
is OK even though X is accessed in two ways. Or did I misunderstand that.

Reply:
Yes, that is ok. I'll change this to say:
> WGSL assumes no aliasing is present between any combination of variables and function parameters. As such, ...

Does that sounds clearer?

Reply:
I think that is better, and addresses @RobinMorisset's point.

I find the way explanation of the rule a bit confusing, so I suggest adding an Issue to revisit the expression of the rule.